### DISCUSSION TITLE
Unexpected output of myst_parser.main.to_html

### DISCUSSION THREAD

**Discussion by pohlt** [executablebooks/MyST-Parser#503] [Q&A]
Let's say I have a very simple md file:
```markdown
```{note}
Hello World
``\` <= without the backslash here, but I don't know how to write it in a code environment
```

If I use `myst-docutils-html5` to convert this, the output is
```html
...
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Hello World</p>
</div>
...
```

I would like to convert it to a simple HTML file (w/o any wrapping or CSS), so I tried `myst_parser.main.to_html`. The (for me unexpected) output is this:
```html
<pre><code class="language-{note}">Hello World
</code></pre>
```

Is it a different default configuration or why is it that `to_html` behaves so differently?

**Comment by john-hen** [comment]
That's not unexpected. The [documentation says](https://myst-parser.readthedocs.io/en/latest/api/parsers.html#parse-myst-markdown-to-html) that `to_html`
> parses markdown and renders as HTML using only the markdown-it parser (i.e. no sphinx or docutils specific processing is done)

So with `to_html` you just get the output of the basic [Markdown-it-py](https://github.com/executablebooks/markdown-it-py) parser, which follows the [CommonMark specification](https://spec.commonmark.org/0.30). For this particular case, the spec says:
> An info string can be provided after the opening code fence. Although this spec doesn’t mandate any particular treatment of the info string, the first word is typically used to specify the language of the code block. In HTML output, the language is normally indicated by adding a class to the `code` element consisting of `language-` followed by the language name.

This is why you get `class="language-{note}"` in the output.

*With* the Docutils-specific processing, which is what `myst-docutils-html5` naturally does, the curly braces around the "language" are a cue to transform that code fence into the corresponding [Doctuils directive](https://docutils.sourceforge.io/docs/ref/rst/directives.html#specific-admonitions), rather than using it for syntax highlighting.

PS: You can use four back-ticks to escape code fences of three back-ticks. That is, this
`````markdown
````markdown
```{note}
Hello World
```
````
````` 
is rendered as this
````markdown
```{note}
Hello World
```
````

**Reply by pohlt** [reply]
Thanks for the quick explanation and the hint (I tried all combinations with backslashes to no avail).
